src/proto/mod.rs

```diff
@@ -40,3 +41,50 @@ impl<'a, T: Serialize + Deserialize<'a>> Blob for T {
         from_bytes(blob)
     }
 }
+
+pub mod recursive {
+    use super::{from_bytes, to_bytes};
+    use serde::{
+        de::{self, Deserializer, Visitor},
+        ser::Serializer,
+        Deserialize, Serialize,
+    };
+    use std::{fmt, marker::PhantomData};
+
+    pub fn serialize<T, S>(obj: &T, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        T: Serialize,
+        S: Serializer,
+    {
+        serializer.serialize_bytes(&to_bytes(obj).unwrap())
+    }
+
+    pub fn deserialize<'de, T, D>(deserialize: D) -> Result<T, D::Error>
+    where
+        T: Deserialize<'de>,
+        D: Deserializer<'de>,
+    {
+        struct RecursiveVisitor<T>(PhantomData<T>);
+
+        impl<'de, T> Visitor<'de> for RecursiveVisitor<T>
+        where
+            T: Deserialize<'de>,
+        {
+            type Value = T;
+
+            fn expecting(&self, formatter: &mut fmt::Formatter) -> fmt::Result {
+                formatter.write_str("an integer between -2^31 and 2^31")
+            }
+
+            fn visit_bytes<E>(self, value: &[u8]) -> Result<Self::Value, E>
+            where
+                E: de::Error,
+            {
+                println!("bytes: {value:?}");
```

Comment:
Please remove this debugging scaffolding :)

Reply:
can you tell I pushed this commit way too late last night? :D

Sorry I'll fix that.